tests/test_solver_utils.py

```diff
@@ -113,3 +116,68 @@ def test_vartype_dimod_smoke(self):
 
         with self.assertRaises(ValueError):
             StructuredSolver.reformat_parameters("INTEGER", {}, {})
+
+class QpuAccessTimeEstimate(unittest.TestCase):
+    """Test QPU access time estimation method."""
+
+    def test_qpu_access_time_estimate_mock(self):
+        solver_mock = StructuredSolver(data=mocks.qpu_pegasus_solver_data(16,
+            problem_timing_data=mocks.qpu_problem_timing_data(qpu='advantage')), client=None)
+
+        # not allowed annealing_time together with anneal_schedule
+        with self.assertRaises(ValueError):
+            solver_mock.estimate_qpu_access_time(num_qubits=1000,
+                                            anneal_schedule=[[0.0, 0.0], [50.0, 0.5]],
+                                            annealing_time=150)
+        # reverse anneal without required initial state
+        with self.assertRaises(ValueError):
+            solver_mock.estimate_qpu_access_time(num_qubits=1000,
+                                            anneal_schedule=[[0.0, 1.0], [2.75, 0.45], [82.75, 0.45], [82.761, 1.0]])
+
+        # currently support is for version 1.0.x
+        with self.assertRaises(ValueError):
+            solver_mock_ver = copy.deepcopy(solver_mock)
```

Comment:
This is fine, but just FYI, [`with mock.patch(...)`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) would fit in nicely here.